### DISCUSSION TITLE
Are you satisfied with the CRD API reference?

### DISCUSSION THREAD

**Discussion by Zeratoxx** [strimzi/strimzi-kafka-operator#7459] [Polls]
Hello folks,

I am working with Strimzi at work. When working with libraries/frameworks/etc., I am preferably using the reference documentation. In my opinion, the Strimzi CRD reference is pretty good regarding the content but I have issues/hickups while browsing through it.

I have two ideas how to improve things: 

1. increase the level of the table of contents
  a. this improves the overwiew
2. extract the reference to a separate documentation page(s)
  a. this reduces the length of the site for visibility and improves navigation

**Comment by scholzj** [comment]
Thanks for providing the feedback, this is useful. My (personal) view -> I would avoid creating separate document for the reference guide. Already today, the books are IMHO confusing and even as a maintainer, I often struggle to know which document book to open for which content. So in my view, I would rather move the other content from the Configuration book into the Deploying book or integrate it directly into the reference guide. WDYT?

As for the ToC => I guess you are using the docs version which is integrated into the website? I.e. https://strimzi.io/docs/operators/latest/configuring.html? Correct? The content there is generated by Javascript and I think at bigger depth, it is too much and it is causing issues. But maybe you can check the separate docs view (i.e. https://strimzi.io/docs/operators/latest/full/configuring.html) => you can get to it by clicking on the _icon_ which is supposed to mean new window. That has a different format and deeper ToC. Would that help?

**Reply by scholzj** [reply]
PS: I pinned it on the Discussions and advertised it on Slack, so that maybe more people find their way here ;-)

**Comment by Zeratoxx** [comment]
>  I would rather move the other content from the Configuration book into the Deploying book or integrate it directly into the reference guide.

I understand your point. In my opinion, combining the contents into one document does not help the issue I guess. I would rather suggest a documentation structure like for Kubernetes itself:
- One website with ...
- ... a superordinated ToC that links ...
- ... multiple subpages which have ...
- ... their own heading structure and thus enable their own detailed ToC and ...
- ... the CRD API reference on a page that is specifically meant to be used for reference documentation.

Well, that would be the icing on the cake, from my point of view, this is best practice. But that would be a lot of work, so I initially suggested ideas with expectedly less work for now 😃

**Reply by scholzj** [reply]
Well, I suspect that Kubernetes has significantly more resources to work on the website and how the docs are presented than we do. Its not just about what we think would be great but also what is the effort to make it happen and to maintain it.

**Reply by Zeratoxx** [reply]
Yeah, sorry, my comment was unfinished 😄

**Reply by Zeratoxx** [reply]
> Its not just about what we think would be great but also what is the effort to make it happen and to maintain it.

You are absolutely right!

**Comment by Zeratoxx** [comment]
>  I guess you are using the docs version which is integrated into the website? I.e. https://strimzi.io/docs/operators/latest/configuring.html? Correct?

Yes.

> you can get to it by clicking on the icon which is supposed to mean new window. That has a different format and deeper ToC. Would that help?

That's very good to know, it definetely helps a lot! Thank you 👍

**Reply by Zeratoxx** [reply]
PS: Sorry, but I have to say, it is pretty unintuitive that the icon link has a different behavior than the text link 😅

**Reply by scholzj** [reply]
> PS: Sorry, but I have to say, it is pretty unintuitive that the icon link has a different behavior than the text link

Well, do you have any better ideas? Some people don't like links opening to a new window outide of the website they are on. So if we link only to the _full_ docs, there will be unhappy users as well 🤷 . Sometiems its hard to make everyone happy. I don't think the current state is ideal, but its IMHO reasonable middleground. 

The main issue if for me that it is a bit unintuitive. I know about it and use almost only the full docs. But I know many people don't realize that the icon leads to something slightly different :-/.

**Comment by Zeratoxx** [comment]
Could be foldable ToCs a solution? 🤔

**Comment by cstmgl** [comment]
Thanks for this, actually I have similar opinion but I would not know how it would become better.
I kind of agree with the point that already now I'm a bit confused in which to find the documentation I'm looking for, so even bigger split might not be better.
I also wanted to say that I was not aware this link existed:
https://strimzi.io/docs/operators/latest/full/configuring.html

It makes browsing much easier.
